hbase-client/src/main/java/org/apache/hadoop/hbase/shaded/protobuf/ProtobufUtil.java

```diff
@@ -3657,9 +3657,10 @@ public static CheckAndMutate toCheckAndMutate(ClientProtos.Condition condition,
           return builder.build((Put) m);
         } else if (m instanceof Delete) {
           return builder.build((Delete) m);
+        } else if (m instanceof Increment) {
+          return builder.build((Increment) m);
         } else {
-          throw new DoNotRetryIOException("Unsupported mutate type: " + mutations.get(0)
-            .getClass().getSimpleName().toUpperCase());
+          return builder.build((Append) m);
```

Comment:
Let's also test for Append and keep the last else branch throwing DoNotRetryIOException?

Reply:
This is for safety. Maybe in the future someone other than us two will implement a new type of Mutation and we will throw a DoNotRetryIOException directly here which will be easy to know what's wrong here, otherwise the behavior will be confusing.

Reply:
Sure. I will add testing for Append and keep the last else branch throwing DoNotRetryIOException.